search and enter without options

panic: runtime error: slice bounds out of range [:-1]

goroutine 1 [running]:
github.com/lucassabreu/clockify-cli/cmd.getProjectID(0x0, 0x0, 0xc000027a20, 0x18, 0xc000080af0, 0xc000027ae0, 0x44de0f, 0x1b, 0xd8b340)
        /home/travis/gopath/src/github.com/lucassabreu/clockify-cli/cmd/common.go:289 +0xa58
github.com/lucassabreu/clockify-cli/cmd.confirmEntryInteractively(0xc000080af0, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xdb5198, ...)
        /home/travis/gopath/src/github.com/lucassabreu/clockify-cli/cmd/common.go:130 +0xa2
github.com/lucassabreu/clockify-cli/cmd.newEntry(0xc000080af0, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xdb5198, ...)
        /home/travis/gopath/src/github.com/lucassabreu/clockify-cli/cmd/common.go:198 +0x439
github.com/lucassabreu/clockify-cli/cmd.glob..func7(0xd7efc0, 0xdb5198, 0x0, 0x0, 0xc000080af0, 0xc00013bcc8, 0x0)
        /home/travis/gopath/src/github.com/lucassabreu/clockify-cli/cmd/in.go:81 +0x2ea
github.com/lucassabreu/clockify-cli/cmd.withClockifyClient.func1(0xd7efc0, 0xdb5198, 0x0, 0x0, 0x0, 0x0)
        /home/travis/gopath/src/github.com/lucassabreu/clockify-cli/cmd/common.go:35 +0x8c
github.com/spf13/cobra.(*Command).execute(0xd7efc0, 0xdb5198, 0x0, 0x0, 0xd7efc0, 0xdb5198)
        /home/travis/gopath/pkg/mod/github.com/spf13/cobra@v1.0.1-0.20200909172742-8a63648dd905/command.go:850 +0x453
github.com/spf13/cobra.(*Command).ExecuteC(0xd811e0, 0x1, 0x0, 0x969f03)
        /home/travis/gopath/pkg/mod/github.com/spf13/cobra@v1.0.1-0.20200909172742-8a63648dd905/command.go:958 +0x349
github.com/spf13/cobra.(*Command).Execute(...)
        /home/travis/gopath/pkg/mod/github.com/spf13/cobra@v1.0.1-0.20200909172742-8a63648dd905/command.go:895
github.com/lucassabreu/clockify-cli/cmd.Execute(0xa15a20, 0x6, 0xa2b520, 0x28, 0xa1a500, 0x14)
        /home/travis/gopath/src/github.com/lucassabreu/clockify-cli/cmd/root.go:51 +0xb2
main.main()
        /home/travis/gopath/src/github.com/lucassabreu/clockify-cli/main.go:26 +0x69

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -11,6 +11,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [v0.16.1] - 2021-06-16
+
 ### Fixed
 
 - interactive selection of project would panic if the list were empty (filtering can empty the list)
@@ -442,7 +444,8 @@ time entry.
 - Golang CLI using [cobra](https://github.com/spf13/cobra)
 - Makefile to help setup actions
 
-[Unreleased]: https://github.com/lucassabreu/clockify-cli/compare/v0.16.0...HEAD
+[Unreleased]: https://github.com/lucassabreu/clockify-cli/compare/v0.16.1...HEAD
+[v0.16.1]: https://github.com/lucassabreu/clockify-cli/releases/tag/v0.16.1
 [v0.16.0]: https://github.com/lucassabreu/clockify-cli/releases/tag/v0.16.0
 [v0.15.1]: https://github.com/lucassabreu/clockify-cli/releases/tag/v0.15.1
 [v0.15.0]: https://github.com/lucassabreu/clockify-cli/releases/tag/v0.15.0

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -11,6 +11,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Changed
+
+- `workspaces` command is now named `workspace`, `workspaces` still supported
+- `workspace` default print format now shows the workspace marked as "default"
+
 ## [v0.16.0] - 2021-05-14
 
 ### Added

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -11,6 +11,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Fixed
+
+- interactive selection of project would panic if the list were empty (filtering can empty the list)
+  and pressing enter. now will return as "no project selected".
+
 ### Changed
 
 - `workspaces` command is now named `workspace`, `workspaces` still supported

--- a/cmd/common.go
+++ b/cmd/common.go
@@ -234,7 +234,7 @@ func newEntry(c *api.Client, te dto.TimeEntryImpl, interactive, allowProjectByNa
 
 func getProjectID(projectID string, workspace string, c *api.Client) (string, error) {
 	projects, err := c.GetProjects(api.GetProjectsParam{
-		Workspace: workspace,
+		Workspace:       workspace,
 		PaginationParam: api.PaginationParam{AllPages: true},
 	})
 
@@ -248,8 +248,8 @@ func getProjectID(projectID string, workspace string, c *api.Client) (string, er
 
 	for i, u := range projects {
 		projectsString[i] = fmt.Sprintf("%s - %s", u.ID, u.Name)
-		if projectNameSize < utf8.RuneCountInString(projectsString[i]) {
-			projectNameSize = utf8.RuneCountInString(projectsString[i])
+		if c := utf8.RuneCountInString(projectsString[i]); projectNameSize < c {
+			projectNameSize = c
 		}
 
 		if found == -1 && u.ID == projectID {
@@ -282,7 +282,7 @@ func getProjectID(projectID string, workspace string, c *api.Client) (string, er
 		projectID = projectsString[found]
 	}
 
-	if projectID, err = ui.AskFromOptions("Choose your project:", projectsString, projectID); err != nil {
+	if projectID, err = ui.AskFromOptions("Choose your project:", projectsString, projectID); err != nil || projectID == "" {
 		return "", nil
 	}
 

--- a/cmd/workspaces.go
+++ b/cmd/workspaces.go
@@ -23,12 +23,14 @@ import (
 
 	"github.com/lucassabreu/clockify-cli/api"
 	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
 )
 
 // workspacesCmd represents the workspaces command
 var workspacesCmd = &cobra.Command{
-	Use:   "workspaces",
-	Short: "List user's workspaces",
+	Use:     "workspace",
+	Aliases: []string{"workspaces"},
+	Short:   "List user's workspaces",
 	RunE: withClockifyClient(func(cmd *cobra.Command, args []string, c *api.Client) error {
 		name, _ := cmd.Flags().GetString("name")
 		format, _ := cmd.Flags().GetString("format")
@@ -41,7 +43,7 @@ var workspacesCmd = &cobra.Command{
 
 		var reportFn func([]dto.Workspace, io.Writer) error
 
-		reportFn = reports.WorkspacePrint
+		reportFn = reports.WorkspacePrint(viper.GetString("workspace"))
 		if format != "" {
 			reportFn = reports.WorkspacePrintWithTemplate(format)
 		}

--- a/reports/workspace.go
+++ b/reports/workspace.go
@@ -19,23 +19,29 @@ func WorkspacePrintQuietly(ws []dto.Workspace, w io.Writer) error {
 }
 
 // WorkspacePrint will print more details
-func WorkspacePrint(ws []dto.Workspace, w io.Writer) error {
-	tw := tablewriter.NewWriter(w)
-	tw.SetHeader([]string{"ID", "Name", "Image"})
+func WorkspacePrint(wDefault string) func(ws []dto.Workspace, w io.Writer) error {
+	return func(ws []dto.Workspace, w io.Writer) error {
+		tw := tablewriter.NewWriter(w)
+		tw.SetHeader([]string{"ID", "Name", "Image"})
 
-	lines := make([][]string, len(ws))
-	for i, w := range ws {
-		lines[i] = []string{
-			w.ID,
-			w.Name,
-			w.ImageURL,
+		lines := make([][]string, len(ws))
+		for i, w := range ws {
+			name := w.Name
+			if wDefault == w.ID {
+				name = name + " (default)"
+			}
+			lines[i] = []string{
+				w.ID,
+				name,
+				w.ImageURL,
+			}
 		}
+
+		tw.AppendBulk(lines)
+		tw.Render()
+
+		return nil
 	}
-
-	tw.AppendBulk(lines)
-	tw.Render()
-
-	return nil
 }
 
 // WorkspacePrintWithTemplate will print each worspace using the format string

